src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java

```diff
@@ -20,41 +22,65 @@
+    public void execute_validIndexUnfilteredOrderList_success() {
         MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
 
         Model expectedModel = model.createCopy();
-        Order orderToBeMarked = expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased())
-                .getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
-        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarked);
+        Order orderToBeMarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarkedInUnfilteredList);
 
-        expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased()).getOrderList()
-                .setOrder(orderToBeMarked, markedOrder);
+        getOrderListByIndex(expectedModel, INDEX_FOURTH_PERSON).setOrder(orderToBeMarkedInUnfilteredList, markedOrder);
         String expectedMessage = String.format(MarkOrderCommand.MESSAGE_MARK_ORDER_SUCCESS,
                 Messages.format(markedOrder));
 
         assertCommandSuccess(markOrderCommand, model, expectedMessage, expectedModel);
     }
 
     @Test
-    public void execute_invalidIndexOrderList_throwsCommandException() {
-        Person person = model.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased());
-        model.updateSelectedPerson(person);
-        Index outOfBoundIndex = Index.fromOneBased(person.getOrderList().size() + 1);
+    public void execute_validIndexFilteredOrderList_success() {
+        personToMarkOrderUnder.updateFilteredOrderList(new StatusEqualsKeywordPredicate(Status.PENDING));
+        MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
+
+        Model expectedModel = model.createCopy();
+        Order orderToBeMarkedInFilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarkedInFilteredList);
+
+        getOrderListByIndex(expectedModel, INDEX_FOURTH_PERSON).setOrder(orderToBeMarkedInFilteredList, markedOrder);
+        String expectedMessage = String.format(MarkOrderCommand.MESSAGE_MARK_ORDER_SUCCESS,
+                Messages.format(markedOrder));
+
+        assertCommandSuccess(markOrderCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
+        MarkOrderCommand markOrderCommand = new MarkOrderCommand(outOfBoundIndex);
+
+        assertCommandFailure(markOrderCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
```

Comment:
the whole point of testing filtered list is to test something that is out of bounds for filtered but not out of bounds before filtered. If not its purpose will be same as that of unfiltered out of bounds. 
maybe you can either copy from me or the ab3?
![image](https://github.com/user-attachments/assets/a1398364-a2cc-47b4-8a91-31343dc3bae9)
![image](https://github.com/user-attachments/assets/a606a1af-ac15-4267-87ed-cf2edd430a54)